src/ripple/shamap/impl/SHAMap.cpp

```diff
@@ -796,7 +796,7 @@ SHAMap::addGiveItem(SHAMapNodeType type, std::shared_ptr<SHAMapItem const> item)
         int branch = selectBranch(nodeID, tag);
         assert(inner->isEmptyBranch(branch));
         auto newNode = makeTypedLeaf(type, std::move(item), cowid_);
-        inner->setChild(branch, newNode);
+        inner->setChild(branch, std::move(newNode));
```

Comment:
Prefer`inner->setChild(makeTypedLeaf(type, std::move(item), cowid_));` which should have optimal semantics.

Reply:
I was trying to make the simplest change that was not changing the semantics, but avoided the copy. Let me look at your suggestion carefully tomorrow.

Reply:
Yes @nbougalis this is cleaner, I just made the change.